pkg/controller/servicemesh/controlplane/gateways_test.go

```diff
@@ -125,21 +125,107 @@ func TestAdditionalIngressGatewayInstall(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "label",
+			smcp: New20SMCPResource(controlPlaneName, controlPlaneNamespace, &v2.ControlPlaneSpec{
+				Gateways: &v2.GatewaysConfig{
+					IngressGateways: map[string]*v2.IngressGatewayConfig{
+						additionalGatewayName: {
+							GatewayConfig: v2.GatewayConfig{
+								Enablement: v2.Enablement{
+									Enabled: &enabled,
+								},
+								Service: v2.GatewayServiceConfig{
+									Metadata: &v2.MetadataConfig{
+										Labels: map[string]string{
+											"test": "test",
+										},
+									},
+								},
+								Namespace: controlPlaneNamespace,
+							},
+						},
+					},
+				},
+			}),
+			create: IntegrationTestValidation{
+				Verifier: VerifyActions(
+					Verify("create").On("networkpolicies").Named("istio-ingressgateway").In(controlPlaneNamespace).Passes(
+						ExpectedLabelMatchedByNetworkPolicy("istio", "ingressgateway"),
+					),
+					Verify("create").On("networkpolicies").Named(additionalGatewayName).In(controlPlaneNamespace).Passes(
+						ExpectedLabelMatchedByNetworkPolicy("test", "test"),
+					),
+					Verify("create").On("deployments").Named(additionalGatewayName).In(controlPlaneNamespace).Passes(
+						ExpectedLabelGatewayCreate("test", "test"),
+					),
+				),
+				Assertions: ActionAssertions{},
+			},
+			delete: IntegrationTestValidation{
+				Assertions: ActionAssertions{
+					Assert("delete").On("deployments").Named(additionalGatewayName).In(controlPlaneNamespace).IsSeen(),
+				},
+			},
+		},
+		{
+			name: "label-2.1",
```

Comment:
this is not great. I will follow-up with a PR that allows to run tests against a set of versions, instead of having to copy-paste the same test with a different version field

Reply:
filed https://issues.redhat.com/browse/MAISTRA-2574